std/experimental/testing/should.d

```diff
@@ -0,0 +1,453 @@
+{
+    if(value !is null) fail("Value is null", file, line);
+}
+
+
+/**
+Verify that the value is not null.
+Throws: UnitTestException on failure
+*/
+void shouldNotBeNull(T)(in T value,
+                     in string file = __FILE__, in ulong line = __LINE__)
+{
+    if(value is null) fail("Value is null", file, line);
+}
+
+/**
+Verify that the value is in the container.
+Throws: UnitTestException on failure
+*/
+void shouldBeIn(T, U)(in T value, in U container,
+                   in string file = __FILE__, in ulong line = __LINE__)
+if(isAssociativeArray!U)
+{
+    if(value !in container)
+    {
+        fail("Value " ~ to!string(value) ~ " not in " ~ to!string(container),
+             file, line);
+    }
+}
+
+/**
+Verify that the value is in the container.
+Throws: UnitTestException on failure
+*/
+void shouldBeIn(T, U)(in T value, in U container,
+                   in string file = __FILE__, in ulong line = __LINE__)
+if(!isAssociativeArray!U)
+{
+    if(find(container, value).empty)
+    {
+        fail("Value " ~ to!string(value) ~ " not in " ~ to!string(container),
+             file, line);
+    }
+}
+
+/**
+Verify that the value is not in the container.
+Throws: UnitTestException on failure
+*/
+void shouldNotBeIn(T, U)(in T value, in U container,
+                      in string file = __FILE__, in ulong line = __LINE__)
+if(isAssociativeArray!U)
+{
+    if(value in container)
+    {
+        fail("Value " ~ to!string(value) ~ " is in " ~ to!string(container),
+             file, line);
+    }
+}
+
+/**
+Verify that the value is not in the container.
+Throws: UnitTestException on failure
+*/
+void shouldNotBeIn(T, U)(in T value, in U container,
+                      in string file = __FILE__, in ulong line = __LINE__)
+if(!isAssociativeArray!U)
+{
+    if(find(container, value).length > 0)
+    {
+        fail("Value " ~ to!string(value) ~ " is in " ~ to!string(container),
+             file, line);
+    }
+}
+
+/**
+Verify that expr throws the templated Exception class.
+Throws: UnitTestException on failure (when expr does not
+throw the expected exception)
+*/
+void shouldThrow(T: Throwable = Exception, E)(lazy E expr,
+                                              in string file = __FILE__,
+                                              in ulong line = __LINE__)
+{
+    if(!threw!T(expr)) fail("Expression did not throw", file, line);
+}
+
+/**
+Verify that expr does not throw the templated Exception class.
+Throws: UnitTestException on failure
+*/
+void shouldNotThrow(T: Throwable = Exception, E)(lazy E expr,
+                                                 in string file = __FILE__,
+                                                 in ulong line = __LINE__)
+{
+    if(threw!T(expr)) fail("Expression threw", file, line);
+}
+
+private bool threw(T: Throwable, E)(lazy E expr)
+{
+    try
+    {
+        expr();
+    }
+    catch(T e)
+    {
+        return true;
+    }
```

Comment:
@burner Yes, shouldThrow will let the exception escape. This is the intended behaviour. The exception with a type other than `T` will be caught by TestCase.collectOutput. As I used `unit-threaded`, silently catching all exceptions was causing me debugging problems, so I made non-expected exceptions be handled slightly differently. In the end, the test still fails (good), but the output gives a stack trace.
@Dicebot Good point.